### PR TITLE
Generic Function Removes Return Type T

### LINKED ISSUE #965: Kdoc fixer false-negative: not generating a Kdoc for several public functions
diktat 1.0.0-RC

```
    fun hasNoChildren() = children.size == 0
    fun getFirstChild() = children.elementAtOrNull(0)
```

```
/**
 * @return
 */
fun hasNoChildren() = children.size == 0
// <------ where is a Kdoc?
fun getFirstChild() = children.elementAtOrNull(0)
```

### LINKED ISSUE #1000: Generic Function Removes Return Type T
## Describe the bug

Not enough information to infer type variable T

Before 
      public suspend inline fun <reified T> get(block: HttpRequestBuilder.() -> Unit): T {
        return httpClient.request(
            base().apply(
                block
            )
        )
    }

After

  public suspend inline fun <reified T> get(block: HttpRequestBuilder.() -> Unit) = httpClient.request(
        base().apply(
            block
        )
    )

## Expected behavior

  public suspend inline fun <reified T> get(block: HttpRequestBuilder.() -> Unit) :T = httpClient.request(
        base().apply(
            block
        )
    )

## Observed behavior

  public suspend inline fun <reified T> get(block: HttpRequestBuilder.() -> Unit) = httpClient.request(
        base().apply(
            block
        )
    )

## Environment information
* diktat version: 1.0.0-rc.2
* build tool (maven/gradle): gradle
* how is diktat run (CLI, plugin, etc.): plugin
* kotlin version: 1.5.21
* operating system: mac
* link to a project (if your project is public): N/A

### PR DESCRIPTION
**Pull request by Cheshiriks**
### What's done:
* fixed bug in WRONG_NEWLINES
* now don't remove return type
Closes #1000

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt [hunk 1]
```diff
@@ -335,18 +335,10 @@ class NewlinesRule(configRules: List<RulesConfig>) : DiktatRule(
                 WRONG_NEWLINES.warnAndFix(configRules, emitWarn, isFixMode,
                     "functions with single return statement should be simplified to expression body", node.startOffset, node) {
                     val funNode = blockNode.treeParent
-                    // if return type is not Unit, then there should be type specification
-                    // otherwise code won't compile and colon being null is correctly invalid
-                    val colon = funNode.findChildByType(COLON)!!
                     val returnType = (funNode.psi as? KtNamedFunction)?.typeReference?.node
-                    val needsExplicitType = returnType != null && (funNode.psi as? KtNamedFunction)?.isRecursive() == true
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:342]: I believe we wanted to make more sophisticated checks for generic classes, didn't we? Also, how will we handle functions with no return type now?

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt (hunk 1 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:342]: OK, let's keep it like this, but please add tests without return type to ensure, that we are not complaining on them

Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:342]: Yes. Added

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt [hunk 2]
```diff
@@ -335,17 +335,11 @@ class NewlinesRule(configRules: List<RulesConfig>) : DiktatRule(
...
                     val expression = node.findChildByType(RETURN_KEYWORD)!!.nextCodeSibling()!!
                     funNode.apply {
-                        if (needsExplicitType) {
-                            removeRange(returnType!!.treeNext, null)
-                        } else {
-                            removeRange(if (colon.treePrev.elementType == WHITE_SPACE) colon.treePrev else colon, null)
+                        if (returnType != null) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:341]: How will the fixer work, when returnType is null? For functions with return type Unit you can specify no explicit return type even with function body. Do we have such tests?
```kotlin
fun foo() {
    println("Logging")
}
```

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt (hunk 2 above)
Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:341]: We don't fix the functions that return Unit

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:341]: We don't fix functions without explicit `return` keyword. But this code is valid:
```kotlin
fun foo() {
    return println("Logging")
}
```
and we will crash in this case

Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:342]: Yes. Fixed this

**Review by petertrr — APPROVED** [review]

**Pull request merged**